frame/identity/src/lib.rs

```diff
@@ -1071,13 +1086,16 @@ decl_module! {
 		///
 		/// The dispatch origin for this call must be _Signed_ and the sender must have a registered
 		/// sub identity of `sub`.
-		#[weight = T::WeightInfo::rename_sub(T::MaxSubAccounts::get())]
-		fn rename_sub(origin, sub: <T::Lookup as StaticLookup>::Source, data: Data) {
+		#[pallet::weight(T::WeightInfo::rename_sub(T::MaxSubAccounts::get()))]
+		pub(super) fn rename_sub(
+			origin: OriginFor<T>, sub: <T::Lookup as StaticLookup>::Source, data: Data
+		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 			let sub = T::Lookup::lookup(sub)?;
 			ensure!(IdentityOf::<T>::contains_key(&sender), Error::<T>::NoIdentity);
 			ensure!(SuperOf::<T>::get(&sub).map_or(false, |x| x.0 == sender), Error::<T>::NotOwned);
 			SuperOf::<T>::insert(&sub, (sender, data));
+			Ok(())
 		}
 
```

Comment:
Not sure why macro required me to add dispatch, just wanted to highlight to make sure this is sensible

Reply:
yes new macro doesn't accept to return nothing, we could enhance the macro though

Reply:
no, lets make returns explicit. less macro magic :)